database/schema.sql

```diff
@@ -0,0 +1,149 @@
+\c hoadb
+DROP TABLE IF EXISTS comment;
+DROP TABLE IF EXISTS vote;
+DROP TABLE IF EXISTS vote_item;
+DROP TABLE IF EXISTS topic;
+DROP TYPE IF EXISTS tag_enum;
+DROP TABLE IF EXISTS member;
+DROP TABLE IF EXISTS hoauser;
+DROP TABLE IF EXISTS community;
+DROP TYPE IF EXISTS state_enum;
+REVOKE ALL PRIVILEGES ON SCHEMA public FROM hoadb;
+\c postgres
+SET ROLE postgres;
+REVOKE ALL PRIVILEGES ON DATABASE hoadb FROM hoadb;
+DROP ROLE IF EXISTS hoadb;
+REVOKE ALL PRIVILEGES ON DATABASE hoadb FROM hoa;
+DROP ROLE IF EXISTS hoa;
+DROP DATABASE IF EXISTS hoadb;
+
+CREATE DATABASE hoadb;
+CREATE ROLE hoadb;
+GRANT CREATE ON DATABASE hoadb TO hoadb;
+-- Initial password will be changed by the app.
+CREATE ROLE hoa LOGIN PASSWORD 'hoa';
+GRANT CREATE ON DATABASE hoadb TO hoadb;
+GRANT CONNECT ON DATABASE hoadb TO hoa;
+\c hoadb
+GRANT USAGE, CREATE ON SCHEMA public TO hoadb;
+SET ROLE hoadb;
+
+-- Supporting just a few states at the moment.
+CREATE TYPE state_enum AS ENUM ('CA', 'AZ', 'NV');
+
+-- An HOA community.
+CREATE TABLE community (
+    id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY, 
+    name VARCHAR(50) UNIQUE NOT NULL, 
+    address VARCHAR(200) NOT NULL,
+    city VARCHAR(100) NOT NULL,
+    state state_enum NOT NULL,
+    zipcode VARCHAR(5) NOT NULL 
+        CONSTRAINT us_zip_code CHECK (regexp_like(zipcode, '^\d{5}$')),
+    icon_file VARCHAR(100));
+
+CREATE INDEX ON community(zipcode);
+
+-- A user registered in hoally. Might be a member of one or more communities.
+CREATE TABLE hoauser (
+    id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    -- Protecting personal information by encrypting.
+    -- We also need the hashed values for querying.
+    hashed_name VARCHAR(200) UNIQUE NOT NULL,
+    encrypted_name VARCHAR(300) NOT NULL,
+    encrypted_full_name VARCHAR(300),
+    hashed_email VARCHAR(200) UNIQUE NOT NULL,
+    encrypted_email VARCHAR(300) NOT NULL,
+    email_validated boolean,
+    -- Only storing hashed passwords or tokens
+    hashed_password VARCHAR(200) NOT NULL,
+    -- Temporary token (user forgot password).
+    hashed_token VARCHAR(200) UNIQUE,
+    token_creation_timestamp TIMESTAMP,
+    encrypted_icon_file VARCHAR(100)
+);
+
+CREATE UNIQUE INDEX ON hoauser(hashed_name, hashed_password);
```

Comment:
This index is not useful. The hashed_name is unique so the implicit index should be good enough.